src/js/web-components/video-player/VideoPlayer.js

```diff
@@ -111,6 +118,47 @@ export default class extends HTMLElement {
       floatingButtonsBar.appendChild(pipButton);
     }
 
+    this.initializeCast().then((castButton) => {
+      floatingButtonsBar.appendChild(castButton);
+
+      window.cast.framework.CastContext.getInstance().addEventListener(
+        window.cast.framework.CastContextEventType.SESSION_STATE_CHANGED,
+        (e) => {
+          if (e.sessionState === 'SESSION_STARTED' || e.sessionState === 'SESSION_RESUMED') {
+            const castSession = window.cast.framework.CastContext.getInstance().getCurrentSession();
+            const mediaInfo = new window.chrome.cast.media.MediaInfo(
+              this.internal.selectedSource.src,
+              this.internal.selectedSource.type,
+            );
+            const metadata = new window.chrome.cast.media.GenericMediaMetadata();
+
+            metadata.title = videoData.title;
+            mediaInfo.metadata = metadata;
+
+            const request = new window.chrome.cast.media.LoadRequest(mediaInfo);
+
+            castSession.loadMedia(request).then(
```

Comment:
Is there any particular reason for not using await there?

Reply:
My reasoning was that since `loadMedia` is the last statement in the block and because the returned promise may reject, it will be a bit more readable to separate the two outcomes without having to `try – catch` the `await` statement and then probably having to `return` early in case of an error or extend the `try` block over subsequent code from the happy path.

```js
// Didn't really want to do this, but looking at it now, it looks more readable than I thought.
try {
  await castSession.loadMedia(request);
} catch (e) {
  console.error(...);
  return;
}

const targetName = castSession.getCastDevice().friendlyName;

if (targetName) {
  this.internal.root.querySelector(`.${CAST_TARGET_NAME}`).innerText = targetName;
  this.classList.add(CAST_HAS_TARGET_NAME);
}
this.classList.add(CAST_CLASSNAME);
```

Reply:
Code below doesn't look too bad as well:

```js
            try {
              await castSession.loadMedia(request);
              const targetName = castSession.getCastDevice().friendlyName;

              if (targetName) {
                this.internal.root.querySelector(`.${CAST_TARGET_NAME}`).innerText = targetName;
                this.classList.add(CAST_HAS_TARGET_NAME);
              }
              this.classList.add(CAST_CLASSNAME);
            } catch(errorCode) {
              /* eslint-disable-next-line no-console */
              console.error(`[Google Cast] Error code: ${errorCode}`),
            }
```

Reply:
@beaufortfrancois Yep, but I usually prefer to go with the shorter `try` block, because it better identifies the statement that can actually throw / reject.

Updated in 2e3d5d9.